src/test/scala/is/hail/io/LoadBgenSuite.scala

```diff
@@ -261,16 +262,42 @@ class LoadBgenSuite extends SparkSuite {
     val sample = "src/test/resources/parallelBgenExport.sample"
 
     hc.indexBgen(bgen)
-    hc.importBgen(bgen, Option(sample)).count()
+    hc.importBgen(bgen, Option(sample), gtField = true, gpField = true, dosageField = false).count()
   }
 
   @Test def testReIterate() {
     hc.indexBgen("src/test/resources/example.v11.bgen")
-    val vds = hc.importBgen("src/test/resources/example.v11.bgen", Some("src/test/resources/example.sample"), contigRecoding = contigRecoding)
+    val vds = hc.importBgen("src/test/resources/example.v11.bgen", Some("src/test/resources/example.sample"),
+      gtField = true, gpField = true, dosageField = false, contigRecoding = contigRecoding)
 
     assert(vds.annotateVariantsExpr("va.cr1 = gs.fraction(g => isDefined(g.GT))")
       .annotateVariantsExpr("va.cr2 = gs.fraction(g => isDefined(g.GT))")
       .rowsTable()
       .forall("va.cr1 == va.cr2"))
   }
+  
+  @Test def testDosage() {
+    for (bgen <- Array("src/test/resources/example.8bits.bgen",
+                       "src/test/resources/example.10bits.bgen")) {
+      if (!hadoopConf.exists(bgen + ".idx"))
+        hc.indexBgen(bgen)
+      
+      val vds = hc.importBgen(bgen, gtField = false, gpField = true, dosageField = true, contigRecoding = contigRecoding)
+        .filterVariantsExpr("va.locus.position == 2000")
+
+      val dosages = vds.queryGenotypes("gs.map(g => g.dosage).collect()")._1
+        .asInstanceOf[IndexedSeq[java.lang.Double]]
+
+      val dosagesFromGP = vds.queryGenotypes("gs.map(g => dosage(g.GP)).collect()")._1
+        .asInstanceOf[IndexedSeq[java.lang.Double]]
+
+      val nNull = dosages.count(_ == null)
+      val nTotal = dosages.length
+      
+      assert(nNull > 0 && nNull < nTotal)
```

Comment:
Can you hardcode the number of null dosages you expect? All of the v1.2 bgen files (*.8bits.*, *.10bits.*) are the same data as example.v11.bgen and example.v11.gen.

Reply:
The number is 2 is the v1.2 files and 1 in the v1 file. I added the v11 file and simplified the last few lines of the test to emphasize that I really just care that there is at least one null value (so it's tested) and that null values occur in the same samples via both routes to dosage.